apps/protocol/lib/lexical/protocol/json_rpc.ex

```diff
@@ -30,11 +31,22 @@ defmodule Lexical.Protocol.JsonRpc do
     {:ok, json_rpc}
   end
 
-  defp do_decode(%{"method" => method, "id" => _id} = request) do
+  defp do_decode(%{"method" => method, "id" => id} = request) do
+    Id.set_latest_request_id(id)
     Requests.decode(method, request)
   end
 
   defp do_decode(%{"method" => method} = notification) do
     Notifications.decode(method, notification)
   end
+
+  defp do_decode(%{"error" => %{"code" => -32_601}}) do
+    # NOTE: vscode can't handle CreateWorkDoneProgress until `Requests.Initialize` is done,
+    # That's why the magic error code is here.
```

Comment:
This is the wrong solution. No one should handle *any* or responses until the initialize response has been sent. Doing work prior to that is a lexical bug.